Tasks/Common/MSBuildHelpers/PathFunctions.ps1

```diff
@@ -5,26 +5,46 @@ function Get-MSBuildPath {
     [CmdletBinding()]
     param(
         [string]$Version,
-        [string]$Architecture)
+        [string]$Architecture,
+        [switch]$SearchCom)
 
     Trace-VstsEnteringInvocation $MyInvocation
     try {
-        $msbuildUtilitiesAssemblies = @(
-            "Microsoft.Build.Utilities.Core, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL"
-            "Microsoft.Build.Utilities.Core, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL"
-            "Microsoft.Build.Utilities.v12.0, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL"
-            "Microsoft.Build.Utilities.v4.0, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL"
-        )
-
-        # Attempt to load a Microsoft build utilities DLL.
-        $index = 0
+        # Attempt to find Microsoft.Build.Utilities.Core.dll from a VS 15 Willow install.
         [System.Reflection.Assembly]$msUtilities = $null
-        while ($msUtilities -eq $null -and $index -lt $msbuildUtilitiesAssemblies.Length) {
-            try {
-                $msUtilities = [System.Reflection.Assembly]::Load((New-Object System.Reflection.AssemblyName($msbuildUtilitiesAssemblies[$index])))
-            } catch [System.IO.FileNotFoundException] { }
+        if ($SearchCom -and
+            ($visualStudio15 = Get-VisualStudio_15_0) -and
+            $visualStudio15.Path) {
 
-            $index++
+            $msbuildUtilitiesPath = [System.IO.Path]::Combine($visualStudio15.Path, "MSBuild\15.0\Bin\Microsoft.Build.Utilities.Core.dll")
+            if (Test-Path -LiteralPath $msbuildUtilitiesPath -PathType Leaf) {
+                Write-Verbose "Loading $msbuildUtilitiesPath"
+                $msUtilities = [System.Reflection.Assembly]::LoadFrom($msbuildUtilitiesPath)
+            }
+        }
+
+        # Fallback to searching the GAC.
+        if (!$msUtilities) {
```

Comment:
This is all existing logic that moved to "fallback" logic (i.e. when not searching COM interface or when no results returned from the COM interface).